azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -98,12 +115,85 @@ public ExecutorManager(Props props, ExecutorLoader loader,
     executingManager = new ExecutingManagerUpdaterThread();
     executingManager.start();
 
+    queueProcessor = new QueueProcessorThread();
+    queueProcessor.start();
+
     long executionLogsRetentionMs =
-        props.getLong("execution.logs.retention.ms",
-            DEFAULT_EXECUTION_LOGS_RETENTION_MS);
+      props.getLong("execution.logs.retention.ms",
+        DEFAULT_EXECUTION_LOGS_RETENTION_MS);
     cleanerThread = new CleanerThread(executionLogsRetentionMs);
     cleanerThread.start();
+  }
 
+  /**
+   * <pre>
+   * Setup activeExecutors using azkaban.properties and database executors
+   * Note:
+   * 1. If a local executor is specified and it is missing from db,
+   *    this method add local executor as active in DB
+   * 2. If a local executor is specified and it is marked inactive in db,
+   *    this method will convert local executor as active in DB
+   * 3. If azkaban.use.multiple.executors is set true, this method will
+   *    load all active projects
+   * </pre>
+   *
+   * @throws ExecutorManagerException
+   */
+  public void setupExecutors() throws ExecutorManagerException {
+    // clear all active executors
+    activeExecutors.clear();
+
+    // Add local executor, if specified as per properties
+    if (azkProps.containsKey("executor.port")) {
+      String executorHost = azkProps.getString("executor.host", "localhost");
+      int executorPort = azkProps.getInt("executor.port");
+      Executor executor =
+        executorLoader.fetchExecutor(executorHost, executorPort);
+      if (executor == null) {
+        executor = executorLoader.addExecutor(executorHost, executorPort);
+      } else if (!executor.isActive()) {
+        executor.setActive(true);
+        executorLoader.updateExecutor(executor);
+      }
+      activeExecutors.add(new Executor(executor.getId(), executorHost,
+        executorPort, true));
+    }
+
+    if (azkProps.getBoolean(AZKABAN_USE_MULTIPLE_EXECUTORS, false)) {
+      activeExecutors.addAll(executorLoader.fetchActiveExecutors());
```

Comment:
seems like the activeExecutors list is build when the webserver is initialized, and after that seems no way to add a new executor to the list without bounce the webserver. I guess it will become a common user scenario to add a new executor to a specific environment and I think to avoid restarting the web box in such event will be really nice.